src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -1780,12 +1780,8 @@ private FeedbackSessionResultsBundle getFeedbackSessionResultsForUserWithParams(
         Map<String, List<FeedbackResponseCommentAttributes>> responseComments = new HashMap<String, List<FeedbackResponseCommentAttributes>>();
         
         //Show all questions even if no responses, unless is an ajax request for a specific question.
-        if (role == UserRole.INSTRUCTOR && !params.containsKey(PARAM_QUESTION_ID)) {
-            for (FeedbackQuestionAttributes question : allQuestions) {
-                relevantQuestions.put(question.getId(), question);
-            }
-        }
-        
+        showAllQuestions(role, params, allQuestions, relevantQuestions);
```

Comment:
In this case, we can either store said boolean in a variable or use a method, albeit much shorter and much simplified and less exposure of what the method "should not know"